trin-core/src/portalnet/types.rs

```diff
@@ -345,6 +345,137 @@ impl FromStr for HexData {
     }
 }
 
+pub enum ContentKey {
+    StateContentKey(StateContentKey),
+    HistoryContentKey(HistoryContentKey),
+}
+
+pub enum StateContentKey {
+  AccountTrieNode(AccountTrieNode),
+  ContractStorageTrieNode(ContractStorageTrieNode),
+  AccountTrieProof(AccountTrieProof),
+  ContractStorageTrieProof(ContractStorageTrieProof),
+  ContractBytecode(ContractBytecode),
+}
+
+impl StateContentKey {
+  // decodes from bytes to ssz container
+  pub fn from_bytes(bytes: &[u8]) -> Result<Self, String> {
+      if let Some(content_type) = bytes.first() {
+          match content_type {
+              0x00 => Ok(StateContentKey::AccountTrieNode(
+                  AccountTrieNode::from_ssz_bytes(&bytes[1..])
+                    .map_err(|e| format!("Failed to decode ssz: {:?}", e))?
+              )),
+              0x01 => Ok(StateContentKey::ContractStorageTrieNode(
+                  ContractStorageTrieNode::from_ssz_bytes(&bytes[1..])
+                    .map_err(|e| format!("Failed to decode ssz: {:?}", e))?
+              )),
+              0x02 => Ok(StateContentKey::AccountTrieProof(
+                  AccountTrieProof::from_ssz_bytes(&bytes[1..])
+                    .map_err(|e| format!("Failed to decode ssz: {:?}", e))?
+              )),
+              0x03 => Ok(StateContentKey::ContractStorageTrieProof(
+                  ContractStorageTrieProof::from_ssz_bytes(&bytes[1..])
+                    .map_err(|e| format!("Failed to decode ssz: {:?}", e))?
+              )),
+              0x04 => Ok(StateContentKey::ContractBytecode(
+                  ContractBytecode::from_ssz_bytes(&bytes[1..])
+                    .map_err(|e| format!("Failed to decode ssz: {:?}", e))?
+              )),
+              _ => Err("No matching content type".to_string()),
+          }
+      } else {
+          Err("No content type".to_string())
```

Comment:
This message be the only text you see in a spew of logs, when trying to debug a problem. Without more context, "No content type" wouldn't help me find the problem easily. Maybe something like this would help a bit more?
```suggestion
          Err("Cannot decode StateContentKey from empty bytes".to_string())
```